alexandria/core/api.py

```diff
@@ -0,0 +1,66 @@
+from django.conf import settings
+from django.core.exceptions import ValidationError
+
+from alexandria.core import models
+
+
+def create_document_file(user, group, document_attributes, file_attributes):
```

Comment:
Since there are some required arguments, it may make sense to have them listed directly in the arguments. That way, it's clear without even needing to document them.

The file arguments could be renamed, so as to show which ones are required as well. Maybe something like this:

```suggestion
def create_document_file(
    user,
    group,
    category,
    title,
    file_name,
    file_content,
    mime_type,
    file_size,
    additional_attributes=None,
    additional_file_attributes=None
):
    additional_attributes = additional_attributes or {}
    additional_file_attributes = additional_file_attributes or {}
```

Reply:
I would suggest renaming the `**attributes` / `**file_attributes` to `**additional_attributes`, to further show that they're *additional* to the regular ones.

Otherwise, the PR looks good now, and ready to merge